ra/registration-authority.go

```diff
@@ -787,9 +789,39 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization
 	}
 
 	// Dispatch to the VA for service
-	ra.VA.UpdateValidations(authz, challengeIndex)
 
-	ra.stats.Inc("RA.UpdatedPendingAuthorizations", 1, 1.0)
+	if !ra.useNewVARPC {
+		ra.VA.UpdateValidations(authz, challengeIndex)
+		ra.stats.Inc("RA.UpdatedPendingAuthorizations", 1, 1.0)
+	} else {
+		go func() {
+			records, err := ra.VA.PerformValidation(authz.Identifier.Value, authz.Challenges[challengeIndex], authz)
+			var prob *probs.ProblemDetails
+			if p, ok := err.(*probs.ProblemDetails); ok {
+				prob = p
+			} else if err != nil {
+				prob = probs.ServerInternal("uncaught error in RPC to VA")
+			}
+
+			// Save the updated records
+			chal := &authz.Challenges[challengeIndex]
+			chal.ValidationRecord = records
+			if prob != nil {
+				chal.Status = core.StatusInvalid
+				chal.Error = prob
+			} else if !chal.RecordsSane() {
+				chal.Status = core.StatusInvalid
+				chal.Error = probs.ServerInternal("Records for validation failed sanity check")
+			} else {
+				chal.Status = core.StatusValid
+			}
+			authz.Challenges[challengeIndex] = *chal
+
+			ra.OnValidationUpdate(authz)
```

Comment:
File a ticket to un-export OnValidationUpdate and remove it from interfaces and RPCs, then add a TODO here.